src/class-wc-gateway-btcpay.php

```diff
@@ -621,8 +621,9 @@ public function process_payment($order_id)
                 throw new \Exception('The BTCPay payment plugin was called to process a payment but could not retrieve the order details for order_id ' . $order_id . '. Cannot continue!');
             }
 
+            $order_number = $order->get_order_number();
```

Comment:
After reading through it and all your comments I understand that the $order_id is the internal order id used to store the order and $order_number is the one that can be manipulated by other plugins and according to the function call it is what is displayed to the user. 

To avoid confusion the quickest solution may be to add a comment above this line saying sth similar to 
// This order number is visible by the user and can differ from the internal $order_id.

Or alternatively maybe renaming one of the variables also solves the confusion? $order_id -> $internal_order_id? (might be confusing as it seems $order_id is widely used in woocommerce, so maybe better do $order_number -> $displayed_order_number?

Reply:
Added with a2bf31f6c8f6f6245bca0956390b33b7c55bb4a6.